### PR TITLE
SELC-2897 Integrated new api to verify if vat number is already registered

### PR DESCRIPTION
**Pull request by OraldoDoci**
#### List of Changes

Integrated new API to verify if the vat number has already been registered for product Fideiussioni.

#### Motivation and Context

During onboarding if the vat number results already registered you should not be able to continue onboarding process as it already exists.

#### How Has This Been Tested?

Run application locally with mocked vat number and interation testing with real API.

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/lib/__mocks__/mockApiRequests.ts [hunk 1]
```diff
@@ -401,6 +401,11 @@ const mockedProducts: Array<Product> = [
     title: 'Carta Giovani',
     status: statusActive,
   },
+  {
+    id: 'prod-fd',
+    title: 'Fideiussioni',
+    status: statusActive,
```
Comment (loremedda) [inline src/lib/__mocks__/mockApiRequests.ts:407]: use correct enum value (ACTIVE)

File: src/hooks/useLogin.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { ENV } from '../utils/env';
 import { UserContext } from '../lib/context';
 
 const testToken =
-  'eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6Imp3dF84Njo3NDoxZTozNTphZTphNjpkODo0YjpkYzplOTpmYzo4ZTphMDozNTo2ODpiNSJ9.eyJlbWFpbCI6ImZ1cmlvdml0YWxlQG1hcnRpbm8uaXQiLCJmYW1pbHlfbmFtZSI6IlNhcnRvcmkiLCJmaXNjYWxfbnVtYmVyIjoiU1JUTkxNMDlUMDZHNjM1UyIsIm5hbWUiOiJBbnNlbG1vIiwiZnJvbV9hYSI6ZmFsc2UsInVpZCI6IjUwOTZlNGM2LTI1YTEtNDVkNS05YmRmLTJmYjk3NGE3YzFjOCIsImxldmVsIjoiTDIiLCJpYXQiOjE2NzQ4MTQxODAsImF1ZCI6ImFwaS5kZXYuc2VsZmNhcmUucGFnb3BhLml0IiwiaXNzIjoiU1BJRCIsImp0aSI6IjAxR1FTQjhLSDNCSks4QkFTRDE4MlZKSDhEIn0.JOfxEC3o8Wor0l430Fq68mWVl4h-NUpFlFuSf6Xgxmu-wqeQyUjRKIfl3M9J0H_8ihxyNMEu5u3PqqQBubGx1mjy24uEsoRPFdLQxlGnpMAM-15SFv8ShDWvMaTSz8hO6vCRJxUtNQgX7SplIG7ZlBBSt7ihwioW1CsKWFISuG0tHwe797NWwaMJlRnzW3R7BIrsGU1eJeue2QqYUnKXZIwYQh21E3EssCNFrusATEuJT_opGaTMzSHpUZxI6cCG2pOE8Cmm0Z75Q2HAM2eoi1_Mx8llZvuk1oVhgDGsACpNRb9Vyxt6jAPUEh3DlkGpLqS8AUD1vRQydzNifiSb4A';
```
Comment (loremedda) [inline src/hooks/useLogin.ts:9]: Reset the test token

File: src/components/steps/StepOnboardingFormData.tsx [hunk 3]
```diff
@@ -362,6 +365,37 @@ export default function StepOnboardingFormData({
     },
   });
 
+  const verifyVatNumber = async () => {
+    setIsVatRegistrated(false);
```
Comment (loremedda) [inline src/components/steps/StepOnboardingFormData.tsx:369]: Why did you set this to false?

**Review by loremedda — CHANGES_REQUESTED** [review]

File: src/components/steps/StepOnboardingFormData.tsx (hunk 3 above)
Comment (OraldoDoci) [reply src/components/steps/StepOnboardingFormData.tsx:369]: To make sure to reset the value to the inital state before the call of the api. This way i dont have old value of setState.

**Review by loremedda — APPROVED** [review]

**Pull request merged**